### DISCUSSION TITLE
Importing markers list

### DISCUSSION THREAD

**Discussion by simonTx1** [g4ixt/QtTinySA#30] [Ideas]
I would like to suggest the ability to create markers manually but also import markers list from a third party software. These wouldn't be dynamic markers that detect peaks on the active traces, like you've implemented already. It would be static markers for visual identification purpose.

One use case : 
Among TinySA users are quite a few RF mics coordinators. 
After assigning a carrier frequency to every RF device, we monitor all of them over 250MHz span and make sure it is not being polluted by any "unauthorized" source. This requires to mark every carrier of interrest (sometimes 200+), with a vertical line. with label and eventually color.
some RF measurment/monitoring software use csv format with this syntax : ID,Frequency,Name,Colour

Regards,

Simon

**Comment by g4ixt** [comment]
Adding more 'functioning' markers is not straightforward because of the way I have coded them as part of the display class - I didn't envisage anyone ever needing more than 4 of them.  But I might be able to add vertical lines on chosen frequencies quite easily.  I will try it in my development branch and see how it looks - I could use the 'frequency bands' in 'preferences' to put a vertical marker at the start and end frequency of each band, for example.  The bands are user-definable and can be added to or removed.

**Comment by g4ixt** [comment]
![Screenshot_20240211_182609](https://github.com/g4ixt/QtTinySA/assets/76836635/06db6fbc-4bd6-4557-abb4-df2666c99a0c)

I have added the ability to have markers associated with the frequency bands.  This means that by adding a frequency band you can create a start and end marker., and turn them on and off   I intend to add more functionality but I will need to update the 'preferences' database to do that - i intend to include being able to create 'bands' table entries (in 'preferences') using the existing markers, by pressing an 'add' button.

This is not yet deployed to github.

**Comment by g4ixt** [comment]
I need a sample CSV file please

**Comment by simonTx1** [comment]
Hi Ian,
Here is a csv marker list from a recent event.
This is an export from my RF mic calculation/coordination software.
This is the format supported by a Spectrum analyzer called "Winradio G33WSM" wich is a reference among RF mics coordinators.

**Reply by g4ixt** [reply]
Thank you, I will use that to test my code

**Comment by g4ixt** [comment]
I have downloaded then removed the CSV file in case the frequencies are confidential.  It would be useful to know if this is the case or if I can leave them in the default database.

**Comment by g4ixt** [comment]
![Screenshot_20240224_090519](https://github.com/g4ixt/QtTinySA/assets/76836635/8ab77d9c-172b-4773-8beb-da55fcaf93ba)
![Screenshot_20240224_091205](https://github.com/g4ixt/QtTinySA/assets/76836635/cbb04557-5ed0-4f23-b8a1-869e4b0aee2a)

Nearly ready to go

**Reply by simonTx1** [reply]
Thanks for your work Ian, it looks great ! I'm looking forward to give it a try.

**Comment by g4ixt** [comment]
Completed and in the v0.10.0 release
